src/AutoMapper/Mappers/StringDictionaryMapper.cs

```diff
@@ -55,8 +55,8 @@ public bool IsMatch(TypePair context)
 
         private static TDestination Map<TDestination>(StringDictionary source, TDestination destination, ResolutionContext context)
         {
-            destination = destination == null ? context.Mapper.CreateObject<TDestination>() : destination;
-            var destTypeDetails = new TypeDetails(typeof(TDestination), _ => true, _ => true);
+            destination = destination == null ? context.Mapper.CreateObject<TDestination>() : destination;
+            var destTypeDetails = new TypeDetails(typeof(TDestination), context.ConfigurationProvider.ConfigurationExpression);
```

Comment:
Those interfaces didn't seem to match. I know you refactored them, but still, maybe you should take a second look :)